### PR TITLE
Add query search and test for item search

### PR DESCRIPTION
**Pull request by fred0405**
Change made:
Add query search with item, and test cases for query searching.

Test:
Screen Shot 2023-04-04 at 5 18 31 PM

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -205,6 +205,7 @@ def test_delete_wishlist(self):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         # make sure the wishlist is deleted
         response = self.app.get(f"{BASE_URL}/{wishlist.id}/items")
+        print(response.get_json())
```
Comment (reubenabraham) [inline tests/test_routes.py:208]: Get rid of the print statement please, thanks

Comment (fred0405) [reply tests/test_routes.py:208]: removed it!

**Review by reubenabraham — CHANGES_REQUESTED** [review]
Can you also update the README with both your query parameter changes? Provide an example and show its working

**Comment by fred0405** [comment]
> Can you also update the README with both your query parameter changes? Provide an example and show its working

Updated readme

**Review by reubenabraham — APPROVED** [review]
LGTM

**Pull request merged**